app/models/user.rb

```diff
@@ -1510,4 +1515,19 @@ def anonymize(new_wca_id = nil)
       country_iso2: special_account? ? country_iso2 : nil,
     )
   end
+
+  def transfer_data_to(new_user)
+    competition_organizers.update_all(organizer_id: new_user.id)
+    competition_delegates.update_all(delegate_id: new_user.id)
+    competitions_results_posted.update_all(results_posted_by: new_user.id)
+    competitions_announced.update_all(announced_by: new_user.id)
+    roles.update_all(user_id: new_user.id)
+    registrations.update_all(user_id: new_user.id)
+
+    return if wca_id.blank?
+
+    wca_id_to_be_transferred = wca_id
```

Comment:
Here is a good place to write `self.wca_id`, to make explicitly clear where this value is coming from